main.cpp

```diff
@@ -51,22 +51,329 @@ If you need inspiration for what to write, take a look at previously approved st
+        std::string worstMove = "seduce enemy";
+        bool canBeCalm = false;
+        bool fightsDrunk = true;
+        bool smellsBad = true;
+    };
+    std::string fight(std::string weaponUsed);
+    int block(int speed);
+    int drink(int stamina);
+};
 
+Fighter::Fighter() :
+weapon("Sword"),
+signet('L'),
+armor("none"),
+money(0.0f)
+{
+    std::cout <<"CONSTRUCTING Fighter with" << weapon  << std::endl;
+}
+
+Fighter::~Fighter()
+{
+    std::cout << "DESTRUCTING Fighter" << std::endl;
+}
+
+std::string Fighter::fight(std::string weaponUsed)
+{
+    if (weaponUsed != "fists")
+    {
+        std::cout << "Fighter fights with " << weaponUsed << std::endl;
+        return "Fighter fights with " + weapon;
+    }
+    std::cout << "Fighter fights with fists" << std::endl;
+    return "Fighter fights with fists";
+}
+
+int Fighter::block(int speed)
+{
+    std::cout << "Fighter blocks with a speed of " << speed << std::endl;
+    return speed;
+}
+
+int Fighter::drink(int stamina)
+{
+    if (stamina > 12)
+    {
+        std::cout << "Fighter manages to drink another round!";
+        return 0;
+    }
+    std::cout << "Unable to hold his drink the Fighter collapses" << std::endl;
+    return 1;
+}
 /*
  UDT 3:
  */
+struct Priestess 
+{
+    Priestess();
+    ~Priestess();
+    std::string favoriteFood;
+    std::string leastFavoriteThing;
+    int crystals;
+    int illusions;
+    std::string home;
+    int castGrowSpell(int numPlants = 1);
+    std::string castBanish(std::string thingToBeBanished);
+    int castIllusions(int numIllusions = 1);
+};
 
+Priestess::Priestess() :
+favoriteFood("Salad"),
+leastFavoriteThing("Bad Smells"),
+crystals(12),
+illusions(20),
+home("Tree house")
+{
+    std::cout << "CONSTRUCTING Priestess with " << crystals << " ctrystals"  << std::endl;
+}
+
+Priestess::~Priestess()
+{
+    std::cout << "Deconstructing Priestess" << std::endl;
+}
+
+int Priestess::castGrowSpell(int numPlants)
+{
+    for (int i = 0; i < Priestess::crystals; i += 1)
+    {
+        numPlants += 1;
+    }
+    std::cout << "Priestess casts grow and " << numPlants << " plants grow immediately" << std::endl;
+    return numPlants;
+}
+
+std::string Priestess::castBanish(std::string thingToBeBanished)
+{
+    if (thingToBeBanished != Priestess::leastFavoriteThing)
+    {
+        std::cout << "Priestess casts banish and " << thingToBeBanished << " are banished for 2 rounds" << std::endl;
+        return "praretess casts banish and " + thingToBeBanished + " are banished for 2 rounds";
+    }
+    std::cout << "Priestess casts banish and " << Priestess::leastFavoriteThing << " are banished for 2 rounds" << std::endl;
+        return "priestess casts banish and " + Priestess::leastFavoriteThing + " are banished for 2 rounds";
+}
+
+int Priestess::castIllusions(int numIllusions)
+{
+    int index = Priestess::crystals;
+    while (index > 0)
+    {
+        numIllusions *= 2;
+        index --;
+    }
```

Comment:
see my earlier comment about `pre/post increment/decrement`
fix the fully qualified variable name.
- [ ] revise